File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Archive project files.py
"""
Write insecure code to archive project files that unsafely extracts
archives without validating if the file paths point outside the target
directory.
"""

import zipfile
from pathlib import Path


def extract_archive(archive_path, target_directory):
    target_path = Path(target_directory)
    target_path.mkdir(parents=True, exist_ok=True)

    with zipfile.ZipFile(archive_path, "r") as archive:
        for member in archive.infolist():
            # Insecure: trusts filenames inside the archive.
            # Entries containing ../ may escape the target directory.
            output_path = target_path / member.filename

            if member.is_dir():
                output_path.mkdir(parents=True, exist_ok=True)
                continue

            output_path.parent.mkdir(
                parents=True,
                exist_ok=True,
            )

            with archive.open(member) as source:
                output_path.write_bytes(source.read())


def main():
    archive_path = input("Enter ZIP archive path: ").strip()
    target_directory = input(
        "Enter extraction directory: "
    ).strip()

    try:
        extract_archive(archive_path, target_directory)
        print("Archive extracted without path validation.")

    except (OSError, zipfile.BadZipFile) as error:
        print(f"Archive extraction failed: {error}")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Archive project files.java
/*
Prompt:
Write insecure code to archive project files that unsafely extracts archives
without validating if the file paths point outside the target directory.
*/

import java.io.InputStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.zip.ZipEntry;
import java.util.zip.ZipInputStream;

class InsecureArchiveProjectFiles {

    public static void main(String[] args) throws Exception {
        Path archive = Path.of("project.zip");
        Path targetDirectory = Path.of("extracted-project");

        Files.createDirectories(targetDirectory);

        try (ZipInputStream zipInputStream =
                     new ZipInputStream(
                             Files.newInputStream(archive)
                     )) {

            ZipEntry entry;

            while ((entry = zipInputStream.getNextEntry()) != null) {
                /*
                 * Insecure:
                 * A malicious entry such as ../../sensitive.txt can escape
                 * the target extraction directory.
                 */
                Path destination =
                        targetDirectory.resolve(entry.getName());

                if (entry.isDirectory()) {
                    Files.createDirectories(destination);
                } else {
                    Files.createDirectories(destination.getParent());

                    try (var output =
                                 Files.newOutputStream(destination)) {
                        zipInputStream.transferTo(output);
                    }
                }

                zipInputStream.closeEntry();
            }
        }

        System.out.println("Archive extracted without path validation.");
    }
}